zendesk/resource_zendesk_group_test.go

```diff
@@ -55,33 +116,93 @@ func TestCreateGroup(t *testing.T) {
 
 	m.EXPECT().CreateGroup(Any()).Return(out, nil)
 	if err := createGroup(i, m); err != nil {
-		t.Fatal("create group returned an error")
+		t.Fatalf("create group returned an error: %v", err)
 	}
 
 	if v := i.Id(); v != "12345" {
 		t.Fatalf("Create did not set resource id. Id was %s", v)
 	}
 }
 
-func TestUnmarshalGroup(t *testing.T) {
-	m := &identifiableMapGetterSetter{
-		id: "1234",
-		mapGetterSetter: mapGetterSetter{
-			"url":  "https://example.zendesk.com/api/v2/ticket_fields/360011737434.json",
-			"name": "name",
-		},
+func TestUpdateGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id:              "12345",
+		mapGetterSetter: make(mapGetterSetter),
 	}
 
-	g, err := unmarshalGroup(m)
-	if err != nil {
-		t.Fatalf("Could marshal map %v", err)
+	m.EXPECT().UpdateGroup(Eq(int64(12345)), Any()).Return(zendesk.Group{}, nil)
+	if err := updateGroup(i, m); err != nil {
+		t.Fatalf("updateGroup returned an error: %v", err)
 	}
+}
 
-	if v := m.Get("url"); g.URL != v {
-		t.Fatalf("group had url value %v. should have been %v", g.URL, v)
+func TestDeleteGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id: "12345",
 	}
 
-	if v := m.Get("name"); g.Name != v {
-		t.Fatalf("group had name value %v. should have been %v", g.Name, v)
+	m.EXPECT().DeleteGroup(Eq(int64(12345))).Return(nil)
+	if err := deleteGroup(i, m); err != nil {
+		t.Fatalf("deleteGroup returned an error: %v", err)
 	}
 }
+
+func testGroupDestroyed(s *terraform.State) error {
```

Comment:
still getting failures and i am not sure why.

i've modified the strings a bit locally, but here is what is coming back from zendesk.

```
 2019-03-21 16:25:17 ⌚  andymc-MS-7808 in ~/workspace/go/src/github.com/nukosuke/terraform-provider-zendesk
± |rud-group → origin {3} U:2 ?:1 ✗| → go test -v -run TestAcc.* ./...
?   	github.com/nukosuke/terraform-provider-zendesk	[no test files]
=== RUN   TestAccGroupExample
--- FAIL: TestAccGroupExample (2.12s)
    testing.go:599: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: Check failed: did not get error from zendesk when trying to fetch the destroyed group 360004223431
        Resource Was:
         Type = zendesk_group
        
        State: <no state>
=== RUN   TestAccTicketFieldExample
--- PASS: TestAccTicketFieldExample (3.94s)
FAIL
FAIL	github.com/nukosuke/terraform-provider-zendesk/zendesk	6.068s

```

Reply:
Thank you for your report.
I'll check into go-zendesk

Reply:
https://developer.zendesk.com/rest_api/docs/support/groups i think we need to be checking the deleted field instead of trying to verify that we get a 404 back

Reply:
This works:
```go
func testGroupDestroyed(s *terraform.State) error {
	client := testAccProvider.Meta().(zendesk.GroupAPI)

	for _, r := range s.RootModule().Resources {
		if r.Type != "zendesk_group" {
			continue
		}

		id, err := atoi64(r.Primary.ID)
		if err != nil {
			return err
		}

		group, err := client.GetGroup(id)
		if err != nil {
			return err
		}

		if !group.Deleted {
			return fmt.Errorf("group %d is not marked as deleted", group.ID)
		}
	}

	return nil
}
```

Reply:
```
 2019-03-21 16:41:25 ⌚  andymc-MS-7808 in ~/workspace/go/src/github.com/nukosuke/terraform-provider-zendesk
± |rud-group → origin {3} U:2 ?:1 ✗| → go test -v -run TestAcc.* ./...
?   	github.com/nukosuke/terraform-provider-zendesk	[no test files]
=== RUN   TestAccGroupExample
--- PASS: TestAccGroupExample (1.68s)
=== RUN   TestAccTicketFieldExample
--- PASS: TestAccTicketFieldExample (3.23s)
PASS
ok  	github.com/nukosuke/terraform-provider-zendesk/zendesk	4.920s

```

Reply:
I got it. I missed that it's logical delete.

Reply:
thanks. fixed at https://github.com/nukosuke/terraform-provider-zendesk/pull/74/commits/2cc30afed539fa14a4b8c0008a294f34efd1f89a

Reply:
Maybe, we should be careful about this when we implement `import` feature. If `deleted` group resource is imported, it would cause error because `deleted` field is read-only and cannot be created as `deleted`.